### PR TITLE
Implement activity tracker

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

Actual implementation of the activity tracker and small docs.
Tested on docker and openshift infrastructures.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            "dom"
+        ],
+        "sourceMap": true,
+        "rootDir": "src",
+        "outDir": "lib"
+    },
+    "include": [
+        "src"
+    ]
+}
```
Comment (benoitf) [inline tsconfig.json]: eof

File: src/node/activity-tracker-service.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { ActivityTrackerService } from "../common/activity-tracker-protocol";
+
+/**
+ * Server side part of Theia activity tracker.
+ * Resieves activity updates from clients and sends reset inactivity requests to Che workspace master.
```
Comment (benoitf) [inline src/node/activity-tracker-service.ts]: typo on Resieves

File: package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "keywords": [
+    "theia-extension"
+  ],
+  "version": "1.0.0",
+  "files": [
+    "lib",
+    "src"
+  ],
+  "dependencies": {
+    "@theia/core": "latest"
```
Comment (benoitf) [inline package.json]: you should specify there fixed versions

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "devDependencies": {
+    "rimraf": "latest",
```
Comment (benoitf) [inline package.json]: you should specify there fixed versions

File: package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "typescript": "latest"
```
Comment (benoitf) [inline package.json]: you should specify there fixed versions

**Comment by benoitf** [comment]
maybe add a travis-ci file to ensure it is compiling fine

File: package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  "files": [
+    "lib",
+    "src"
+  ],
+  "dependencies": {
+    "@theia/core": "0.3.14"
+  },
+  "devDependencies": {
+    "rimraf": "2.6.2",
+    "typescript": "2.8.4"
```
Comment (AndrienkoAleksandr) [inline package.json]: Maybe it would be better to use one https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16945 .

Comment (mmorhun) [reply package.json]: Yes, fixed in main PR.

**Comment by mmorhun** [comment]
Closing this as temporary place.
Persistant repository for the extension is https://github.com/eclipse/che-theia-activity-tracker

**Pull request closed without merging**